docs/BUILD.bazel

```diff
@@ -1,10 +0,0 @@
-# This load statement must be in the docs/ package rather than anything users depend on
```

Comment:
I like the idea of including docs in the release and displaying them on the BCR. That also neatly solves versioned documentation. Nonetheless, there's also value to having documentation available for HEAD either for contributors or for users that need to override a dependency to an unreleased version, e.g. because they depend on a non-released feature. Is this new approach incompatible with `stardoc_with_diff_test`? Or would `stardoc_with_diff_test` still work with the `bazel_lib` version of `bzl_library`?